File: CS/Controllers/OrdersController.cs
using DevExpress.Web.Mvc;
using DevExpress.Xpo;
using System.Linq;
using System.Web.Mvc;

namespace DevExpressMvcApplication.Controllers {
    public class OrdersController : BaseXpoController<Order> {
        public ActionResult Index() {
            return View("Index", GetOrders());
        }

        public ActionResult IndexPartial() {
            return PartialView("IndexPartial", GetOrders());
        }

        [HttpPost]
        public ActionResult EditOrder([ModelBinder(typeof(DevExpressEditorsBinder))] OrderViewModel order) {
            Save(order);
            return PartialView("IndexPartial", GetOrders());
        }

        [HttpPost]
        public ActionResult DeleteOrder([ModelBinder(typeof(DevExpressEditorsBinder))] OrderViewModel order) {
            Delete(order);
            return PartialView("IndexPartial", GetOrders());
        }

        OrdersViewModel GetOrders() {
            return new OrdersViewModel() {
                Source = (from o in XpoSession.Query<Order>().ToList()
                            select new OrderViewModel() {
                                ID = o.Oid, Name = o.Name, Customer = o.Customer.Oid
                            }).ToList(),
                CustomersLookUpData = (from c in XpoSession.Query<Customer>().ToList()
                                        select new CustomerViewModel() {
                                            ID = c.Oid, Name = c.Name
                                        }).ToList()
                };
        }
    }
}

File: VB/Controllers/OrdersController.vb
Imports DevExpress.Web.Mvc
Imports DevExpress.Xpo
Imports System.Linq
Imports System.Web.Mvc

Namespace DevExpressMvcApplication.Controllers
	Public Class OrdersController
		Inherits BaseXpoController(Of Order)

		Public Function Index() As ActionResult
			Return View("Index", GetOrders())
		End Function

		Public Function IndexPartial() As ActionResult
			Return PartialView("IndexPartial", GetOrders())
		End Function

		<HttpPost>
		Public Function EditOrder(<ModelBinder(GetType(DevExpressEditorsBinder))> ByVal order As OrderViewModel) As ActionResult
			Save(order)
			Return PartialView("IndexPartial", GetOrders())
		End Function

		<HttpPost>
		Public Function DeleteOrder(<ModelBinder(GetType(DevExpressEditorsBinder))> ByVal order As OrderViewModel) As ActionResult
			Delete(order)
			Return PartialView("IndexPartial", GetOrders())
		End Function

		Private Function GetOrders() As OrdersViewModel
			Return New OrdersViewModel() With {.Source = (
			    From o In XpoSession.Query(Of Order)().ToList()
			    Select New OrderViewModel() With {.ID = o.Oid, .Name = o.Name, .Customer = o.Customer.Oid}).ToList(), .CustomersLookUpData = (
			    From c In XpoSession.Query(Of Customer)().ToList()
			    Select New CustomerViewModel() With {.ID = c.Oid, .Name = c.Name}).ToList()}
		End Function
	End Class
End Namespace